### ISSUE TITLE
can‘t perform step3

### ISSUE THREAD

**Issue by ge1mina023** [Daniil-Osokin/lightweight-human-pose-estimation.pytorch#212]
I was training the model according to step3.
my pytorch version is 1.8.0 and I use my M1 Macbook for this training

when I input the command of step 3 like:  **python train.py --train-images-folder <MY_COCO_HOME>/train2017/ --prepared-train-labels prepared_train_annotation.pkl --val-labels val_subset.json --val-images-folder <MY_COCO_HOME>/val2017/ --checkpoint-path <my_path_to>/mobilenet_sgd_68.848.pth.tar --from-mobilenet**

I get the message as follow:
**[WARNING] Not found pre-trained parameters for model.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.1.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.2.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.2.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.3.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.3.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.4.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.4.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.5.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.5.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.6.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.6.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.7.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.7.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.8.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.8.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.9.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.9.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.10.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.10.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.11.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for model.11.4.num_batches_tracked
[WARNING] Not found pre-trained parameters for cpm.align.0.weight
[WARNING] Not found pre-trained parameters for cpm.align.0.bias
[WARNING] Not found pre-trained parameters for cpm.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.0.2.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.1.2.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.2.0.weight
[WARNING] Not found pre-trained parameters for cpm.trunk.2.2.weight
[WARNING] Not found pre-trained parameters for cpm.conv.0.weight
[WARNING] Not found pre-trained parameters for cpm.conv.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.trunk.2.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.trunk.2.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.heatmaps.0.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.heatmaps.0.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.heatmaps.1.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.heatmaps.1.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.pafs.0.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.pafs.0.0.bias
[WARNING] Not found pre-trained parameters for initial_stage.pafs.1.0.weight
[WARNING] Not found pre-trained parameters for initial_stage.pafs.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.0.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.1.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.2.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.3.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.initial.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.initial.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.0.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.running_mean
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.running_var
[WARNING] Not found pre-trained parameters for refinement_stages.0.trunk.4.trunk.1.1.num_batches_tracked
[WARNING] Not found pre-trained parameters for refinement_stages.0.heatmaps.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.heatmaps.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.heatmaps.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.heatmaps.1.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.pafs.0.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.pafs.0.0.bias
[WARNING] Not found pre-trained parameters for refinement_stages.0.pafs.1.0.weight
[WARNING] Not found pre-trained parameters for refinement_stages.0.pafs.1.0.bias**

after that, I did not get any message like someone else, i.e. **Validation...**
I've checked all the previous steps and generated everything,but,this confused state remained for several hours.
Anything else I missed?

After my debugging, it got stuck at the step of **for batch_data in train_loader:** in the **train.py**

**Comment by Daniil-Osokin** [comment]
Hi, warnings are ok, check #93. Stuck in the `train_loader` looks strange (are you using COCO dataset?), you can set `num_workers=0` and try to find the line which causes the stuck.

**Comment by ge1mina023** [comment]
@Daniil-Osokin Inlcuding COCO dataset, everything is rational. For this conclusion, I checked many times.

**Comment by ge1mina023** [comment]
@Daniil-Osokin Inlcuding COCO dataset, everything is rational. For this conclusion, I checked many times.

**Comment by Daniil-Osokin** [comment]
To dig into reasons, I suggest to set `num_workers=0` and find the line inside the dataset, which causes the stuck.

**Comment by Daniil-Osokin** [comment]
Hope, it helped.
